openshift/templates/README.md

```diff
@@ -0,0 +1,137 @@
+
+### Through the OpenShift UI or otherwise, you should to create:
+
+OpenShift secret named ```mongo-secrets``` with key:
+```
+MONGO_ADMIN_PASSWORD
+
+```
+that is the password for your mongo admin account on your live database
+
+OpenShift config map named ```mongo-config``` with keys:
+```
+MONGO_BACKUP_COUNT
+```
+that is the number of backups you want to keep in your PVC
+```
+MONGO_BACKUP_SIZE_LOWER_LIMIT
+```
+that is the lower size limit in bytes that your backups should be above to consider them successful
+```
+MONGO_BACKUP_SIZE_UPPER_LIMIT
+```
+that is the upper size limit in bytes that your backups should be above to consider them successful
+```
+ROCKETCHAT_BACKUP_DB_WEBHOOK
+```
+that is the rocketchat webhook for your backup alert channel
+### To create the cronjob:
+
+Copy the login command from the OpenShift UI and paste it into your console
+
+Navigate to your project:
+
+```
+oc project <your project>
+```
+
+cd into this directory (or wherever your template-mongodb-backup.yaml file is)
+
+Create the cronjob:
+
+```
+oc process -f template-mongodb-backup.yaml MONGODB_BACKUP_VOLUME_CLAIM=bk-mem-mmt-prod-glglwmmwtkmg MONGODB_BACKUP_SCHEDULE='30 6 * * *' | oc create -f -
+```
+
+where ```MONGODB_BACKUP_VOLUME_CLAIM``` is the PVC that your backups will be sored in and ```MONGODB_BACKUP_SCHEDULE``` is the cron scedule that this job will run on (30 6 * * * is every day at 6:30 AM UCT which is 11:30 PM PCT)
+
+## How to Restore a Mongo Database
+
+### Mount your backup PVC
+
+mount your backup PVC to a container so that you can access your backups. I use PVC migrator in this example (https://github.com/BCDevOps/StorageMigration/tree/master/openshift/templates) but you can use anything with a mount point. there should be a build of PVC migrator in the tools namespace so you should just have to deploy it in your working namespace from there.
+
+In the OpenShift UI, on the deployments page click "Add to Project" then "Import YAML/ JSON" and paste the deployment config in there (https://raw.githubusercontent.com/BCDevOps/StorageMigration/master/openshift/templates/pvc-migrator-deploy.yaml). Input the correct values, deploy and spin up a pod.
+
+### Copy your dump onto your Mongo pod
+
+After you have mounted your mongo backup PVC to a container and spun up a pod, you can copy your backup onto the mongo pod and restore from there.
+
+First figure out which is the last healthy backup and copy that file name (e.g. dump-2019-04-15-17:14:15-HEALTHY). To do this, rsh into the pod you just created and look in the mount path (/source in this example).
+
+
+copy your dump locally using rsync
+```
+oc rsync <backup pod>:/<mount directory>/<healthy backup folder> .
+```
+e.g.
+```
+oc rsync pvc-migrator-2-b7tf5:/source/dump-2019-04-16-18:29:23-HEALTHY .
+```
+
+cd into your dump directory (note you cannot rsync the dump-2019-04-16-18:29:23-HEALTHY dump folder name as it is, because it has colons in it, and rsync will recognise it as a pod)
```

Comment:
Looks like line 178 of openshift/templates/template-mongodb-backup.yaml uses dashes now so perhaps this is no longer relevant?